prov/gni/src/gnix_rma.c

```diff
@@ -816,11 +819,13 @@ static void __gnix_rma_more_fill_pd(struct gnix_fab_req *req,
 					 sizeof(gni_ct_put_post_descriptor_t));
 		more_put = (gni_ct_put_post_descriptor_t *)
 			   txd->gni_more_ct_descs;
+		assert(more_put);
```

Comment:
This assert (as well as the one below) would be more valuable below where you use more_get and more_put.